src/nose2_contrib/jira/callbacks.py

```diff
@@ -5,62 +5,68 @@
-            raise KeyError("{} does not exist, please register it.".format(name))
-        return cls.registry[name]
-
-
-@JiraRegistry.register('write_success_comment', False)
-def write_success_comment(jira_plugin, jira_issue, test, message):
+def add_comment(jira_plugin, jira_issue, test, message, *, message_format):
     """
     Write comment to notify test success.
 
     :param jira_issue: the jira issue object
     :param test: the test case
     :param message: the success message
+    :param message_format: the message format. As of 1.0 we use ``str.format`` syntax and accepted keys are ``test``
+    and ``message``
     """
     if not jira_plugin.connected:
         return
-    jira_plugin.jira_client.add_comment(jira_issue, message)
-    jira_plugin.logger.info("Success comment sent to {jira_issue.id} for {test}".format(jira_issue=jira_issue, test=test))
+    jira_plugin.jira_client.add_comment(jira_issue, message_format.format(test=test, message=message))
+    jira_plugin.logger.info("Success comment sent to {jira_issue.id} for {test}".format(jira_issue=jira_issue,
+                                                                                        test=test))
+
+JiraRegistry.register('write_success_comment', False,
+                      message_format="{test} has successed.")(add_comment)
 
 
-@JiraRegistry.register('write_failure_and_back_in_dev', False)
-def write_failure_and_back_in_dev(jira_plugin, jira_issue, test, message):
+def apply_jira_transition(jira_plugin, jira_issue, test, message, *, jira_transition, message_format):
     """
     report a failure to jira and send back to dev.
 
     :param jira_issue: the jira issue object
     :type jira_issue: jira.resources.Issue
     :param test: the testcase object
     :param message: the message to send to jira
+    :param jira_transition: The transition to apply
+    :param message_format: the transition message format. As of 1.0 we use ``str.format`` syntax and accepted keys are
+    ``test`` and ``message``. To bypass "transition message submission" give ``None``
     """
     if not jira_plugin.connected:
         return
-    jira_plugin.jira_client.add_comment(jira_issue, "Automated tests {} failed with messages :\n {}".format(test, message))
-    jira_plugin.logger.info("Failure comment sent to {jira_issue.id} for {test}".format(jira_issue=jira_issue, test=test))
-    transition_id = jira_plugin.jira_client.find_transitionid_by_name(jira_issue, 'Set as To Do')
+    if message_format:
+        add_comment(jira_plugin, jira_issue, test, message, message_format=message_format)
+    transition_id = jira_plugin.jira_client.find_transitionid_by_name(jira_issue, jira_transition)
     jira_plugin.jira_client.transition_issue(jira_issue, transition_id)
 
 
+def register_transition(registration_name, jira_transition, transition_message_format):
+    """
+    register a transition to support for your test. Example, To send back a ticket to development you can probably use
```

Comment:
"to support for your test" unclear....  Maybe just "Register a ticket transition"?  Unclear what this is doing exactly.

Reply:
I will try to make it clearer.